app/controllers/application_controller.rb

```diff
@@ -171,7 +171,7 @@ def initialize_service_request
 
   def authorize_identity
     # If the request is in first_draft status
-    if @service_request.status == 'first_draft' && (action_name == 'catalog' || (Rails.application.routes.recognize_path(request.referrer)[:action] == 'catalog' && request.format.js?))
+    if @service_request.status == 'first_draft' && (action_name == 'catalog' || (Rails.application.routes.recognize_path(request.referrer)[:action] == 'catalog' && (request.format.js? || request.format.json?)))
```

Comment:
Actually I'm wondering if it would be better to do `!request.format.html?` to cover all cases?